pynest/examples/Potjans_2014/network.py

```diff
@@ -19,486 +19,522 @@
+        self.ext_indegrees = np.round((self.net_dict['K_ext'] *
+                                       self.net_dict['K_scaling'])).astype(int)
+
+        # conversion from PSPs to PSCs
+        mean_PSC_matrix = helpers.weight_as_current_from_potential(
+            self.net_dict['mean_PSP_matrix'],
+            self.net_dict['neuron_params']['C_m'],
+            self.net_dict['neuron_params']['tau_m'],
+            self.net_dict['neuron_params']['tau_syn'])
+        PSC_ext = helpers.weight_as_current_from_potential(
+            self.net_dict['PSP_e'],
+            self.net_dict['neuron_params']['C_m'],
+            self.net_dict['neuron_params']['tau_m'],
+            self.net_dict['neuron_params']['tau_syn'])
+
+        # DC input compensates for potentially missing Poisson input
+        if self.net_dict['poisson_input']:
+            DC_amp = np.zeros(self.num_pops)
+        else:
+            if nest.Rank() == 0:
+                print('DC input compensates for missing Poisson input.\n')
+            DC_amp = helpers.dc_input_compensating_poisson(
+                self.net_dict['bg_rate'], self.net_dict['K_ext'],
+                self.net_dict['neuron_params']['tau_syn'],
+                PSC_ext)
+
+        # adjust weights and DC amplitude if the indegree is scaled
+        if self.net_dict['K_scaling'] != 1:
+            mean_PSC_matrix, PSC_ext, DC_amp = \
+                helpers.adjust_weights_and_input_to_synapse_scaling(
+                    self.net_dict['full_num_neurons'],
+                    full_num_synapses, self.net_dict['K_scaling'],
+                    mean_PSC_matrix, PSC_ext,
+                    self.net_dict['neuron_params']['tau_syn'],
+                    self.net_dict['full_mean_rates'],
+                    DC_amp,
+                    self.net_dict['poisson_input'],
+                    self.net_dict['bg_rate'], self.net_dict['K_ext'])
+
+        # store final parameters as class attributes
+        self.mean_weight_matrix = mean_PSC_matrix
+        self.std_weight_matrix = self.net_dict['std_PSP_matrix']
+        self.weight_ext = PSC_ext
+        self.DC_amp = DC_amp
+
+        # thalamic input
+        if self.stim_dict['thalamic_input']:
+            num_th_synapses = helpers.num_synapses_from_conn_probs(
+                self.stim_dict['conn_probs_th'],
+                self.stim_dict['num_th_neurons'],
+                self.net_dict['full_num_neurons'])[0]
+            self.weight_th = helpers.weight_as_current_from_potential(
+                self.stim_dict['PSP_th'],
+                self.net_dict['neuron_params']['C_m'],
+                self.net_dict['neuron_params']['tau_m'],
+                self.net_dict['neuron_params']['tau_syn'])
+            if self.net_dict['K_scaling'] != 1:
+                num_th_synapses *= self.net_dict['K_scaling']
+                self.weight_th /= np.sqrt(self.net_dict['K_scaling'])
+            self.num_th_synapses = num_th_synapses.astype(int)
```

Comment:
Here is still a 'floor-cast-to-int' instead of the `np.round(x).astype(int)`

Reply:
Thanks. It is fixed now.